src/include/libpmemobj/ctl.h

```diff
@@ -153,6 +153,12 @@ struct pobj_alloc_class_desc {
 	 * The identifier of this allocation class.
 	 */
 	unsigned class_id;
+
+	/*
+	 * Desired alignment of objects from the allocation class.
+	 * If non zero, must be an even divisor of unit size.
+	 */
+	size_t alignment;
```

Comment:
You can't just add new fields to public structures, without some kind of versioning.

Application compiled against older version will not initialize this field, so newer pmemobj will get garbage here (in the best case) or will crash (if end of structure is aligned to page boundary and there's nothing mapped after).

Reply:
Yes, I understand that. Solutions?

I'm currently thinking about simply adding "size_t alignment;" field as unused to 1.4 release, so that we don't have this problem.
Different solution would be to create a second CTL entry point with the new version.

Reply:
IMO the simplest solution is to add "version" or "size" field in 1.4.